code/modules/power/gravitygenerator.dm

```diff
@@ -132,7 +132,13 @@
 	return ..()
 
 /obj/machinery/gravity_generator/main/proc/eventshutofftoggle() // Used by the gravity event. Bypasses charging and all of that stuff.
+	breaker = 0
 	set_state(eventon)
+	sleep(20)
+	charge_count = 0
```

Comment:
How long does this take to charge back up? Is it comparable to the old event duration?

Reply:
However long it takes for the gravity generator to normally turn on. It is not comparable as the old event ended within seconds of starting.